modules/contact/src/ContactAction.C

```diff
@@ -90,7 +90,8 @@ ContactAction::act()
       InputParameters params = _factory.getValidParams("MechanicalContactConstraint");
 
       // Extract global params
-      _app.parser().extractParams(_name, params);
+      if (isParamValid("parser_syntax"))
+        _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
That's true but these Actions are already broken which is why they are manually extracting information. If you look up above there's a conditional followed by the retrieval of a sub-InputParameters object. It's being dynamically created and filled on the fly. This is very wrong today with the capabilities of the Action system today. It's not possible to fill all of the parameters needed by these sub-InputParameters with the existing logic so it just simply won't work if parser_syntax _doesn't_ exist.

I'm only pointing out that your logic isn't equivalent so you should throw an error instead. The correct way to fix these actions would be to associate them with multiple blocks and multiple tasks so that the parser can extract all information for them.